xpass/xpass.cc

```diff
@@ -220,12 +249,35 @@ void XPassAgent::recv_credit_stop(Packet *pkt) {
   credit_send_state_ = XPASS_SEND_CLOSED;
 }
 
-void XPassAgent::handle_retransmit() {
+void XPassAgent::handle_sender_retransmit() {
   switch (credit_recv_state_) {
     case XPASS_RECV_CREDIT_REQUEST_SENT:
       send(construct_credit_request(), 0);
-      retransmit_timer_.resched(retransmit_timeout_);
+      sender_retransmit_timer_.resched(retransmit_timeout_);
+      break;
+    case XPASS_RECV_CREDIT_STOP_SENT:
+      credit_recv_state_ = XPASS_RECV_CLOSE_WAIT;
+      credit_recved_ = 0;
+      sender_retransmit_timer_.resched(retransmit_timeout_); 
```

Comment:
This also should use rtt_ or default_credit_stop_timeout_ instead of retransmit_timeout_. In this case, we can use just rtt_ not 2 * rtt_.